packages/libs/error-stack/src/ser.rs

```diff
@@ -0,0 +1,152 @@
+//! Implementation of general report serialization.
+//!
+//! The value can be of any type, currently only printable attachments and context are supported, in
+//! the near future any values will be supported through the use of hooks.
+//!
+//! The returned JSON returned is a list of all current sources with the following output:
+//!
+//! ```json
+//! {
+//!     "context": "string value",
+//!     "attachments": ["all attachments leading up to this context"],
+//!     "sources": [] // recursive render using `frame.sources()`
+//! }
+//! ```
+
+use alloc::{format, vec, vec::Vec};
+
+use serde::{ser::SerializeMap, Serialize, Serializer};
+
+use crate::{AttachmentKind, Context, Frame, FrameKind, Report};
+
+struct SerializeAttachment<'a>(&'a Frame);
```

Comment:
Is there a specific reason we don't implement `Serialize` for `Frame`? I could see value in this being used directly. This could also be done later.

Reply:
Yes, because we cannot guarantee that `Frame` serializes correctly. In that case, we would need a fallback (like `opaque`) type of frame.